backend/data.go

```diff
@@ -24,10 +25,17 @@ type DataQuery struct {
 	JSON          json.RawMessage
 }
 
-// QueryDataResponse holds the results for a given query.
+// QueryDataResponse contains the results from a QueryDataRequest.
 type QueryDataResponse struct {
-	Frames   []*data.Frame
-	Metadata map[string]string
+	Responses map[string]*DataResponse
```

Comment:
Not sure if I should make `map[string]*DataResponse` a type (`DataResponses`?). I could then make make a New method for it, so consumers are don't have to `foo := map[string]*DataResponse{}` or `make`.

Could also then add a func that returns a DataResponses from the queries (setting the refid and zero-valued struct for you).